### PR TITLE
Accept PURLs for package submission

### LINKED ISSUE #1336: Extension API: Submit by PURL
`PhylumApi.analyze()` should support submitting dependencies as PURLs (just like the REST API [does][1])

[1]: https://api.phylum.io/api/v0/swagger/index.html#/Jobs/jobs_start_job

_Originally posted by @kylewillmon in https://github.com/phylum-dev/community-extensions/pull/33#discussion_r1449261706_

### PR DESCRIPTION
**Pull request by cd-work**
This patch changes the `api::submit_request` function to accept PURLs with optional dependency file in addition to the existing package descriptors.

Closes #1336.

### CODE REVIEW

File: extensions/phylum.ts
Comment (maxrake) [inline extensions/phylum.ts:1]: Are the changes here enough to warrant an entry in `extensions/CHANGELOG.md`?

File: cli/src/commands/extensions/api.rs [hunk 1]
```diff
@@ -151,7 +164,7 @@ struct ProcessOutput {
 #[serde]
 async fn analyze(
     op_state: Rc<RefCell<OpState>>,
-    #[serde] packages: Vec<PackageWithOrigin>,
+    #[serde] packages: Vec<ExtensionApiPackage>,
```
Comment (maxrake) [inline cli/src/commands/extensions/api.rs:167]: Is this change worthy of a `CHANGELOG.md` entry?

**Review by maxrake — APPROVED** [review]
The code was reviewed with a non-Rusty eye and appears to be okay. The changes were not tested locally. There are some questions about CHANGELOG entries.

File: cli/src/commands/extensions/api.rs (hunk 1 above)
Comment (cd-work) [reply cli/src/commands/extensions/api.rs:167]: Yes we should state that purl is now a supported format.

**Review by maxrake — CHANGES_REQUESTED** [review]
`deno fmt` change requested but otherwise LGTM.

**Pull request merged**